src/operator/tensor/elemwise_unary_op.h

```diff
@@ -252,6 +252,39 @@ class UnaryOp : public OpBase {
     });
   }
 
+  template<typename xpu, typename OP>
+  static void ComputeMixedType(const nnvm::NodeAttrs& attrs,
+                      const OpContext& ctx,
+                      const std::vector<TBlob>& inputs,
+                      const std::vector<OpReqType>& req,
+                      const std::vector<TBlob>& outputs) {
+    mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
+
+    if (mxnet::common::is_float(inputs[0].type_flag_)) {
+      MSHADOW_REAL_TYPE_SWITCH(outputs[0].type_flag_, DType, {
+        MSHADOW_REAL_TYPE_SWITCH(inputs[0].type_flag_, IType, {
```

Comment:
this is the case when forward computation's input is floating point number, the output will be of the same type as the input, so you can directly call the original `Compute` function above, no need for launching the kernel here.